lib/emarsys/params_converter.rb

```diff
@@ -9,22 +9,37 @@ def initialize(params={})
     end
 
     def convert_to_ids
+      params.is_a?(Hash) ? convert_hash_to_ids : convert_array_to_ids
+    end
+
+    def convert_to_identifiers
       new_hash = {}
       params.each do |key, value|
-        matching_attributes = Emarsys::FieldMapping::ATTRIBUTES.find{|elem| elem[:identifier] == key.to_s}
-        new_hash.merge!({ (matching_attributes.nil? ? key : matching_attributes[:id]) => value })
+        matching_attributes = Emarsys::FieldMapping::ATTRIBUTES.find{|elem| elem[:id] == key.to_i && key.to_i != 0}
+        new_hash.merge!({ (matching_attributes.nil? ? key : matching_attributes[:identifier]) => value })
       end
       new_hash
     end
 
-    def convert_to_identifiers
+    private
+
+    def convert_hash_to_ids
       new_hash = {}
       params.each do |key, value|
-        matching_attributes = Emarsys::FieldMapping::ATTRIBUTES.find{|elem| elem[:id] == key.to_i && key.to_i != 0}
-        new_hash.merge!({ (matching_attributes.nil? ? key : matching_attributes[:identifier]) => value })
+        matching_attributes = Emarsys::FieldMapping::ATTRIBUTES.find{|elem| elem[:identifier] == key.to_s}
+        new_hash.merge!({ (matching_attributes.nil? ? key : matching_attributes[:id]) => value })
       end
       new_hash
     end
```

Comment:
The diff here looks weird. I actually didn't change the `convert_to_identifiers` method, just added two private methods.